### DISCUSSION TITLE
New software

### DISCUSSION THREAD

**Discussion by 25Tim** [GrandOrgue/OdfEdit#63] [Ideas]
@eturpault
Hello Eric,
have you interest for programming a new Software? For writing Pitch-informations in Hauptwerk-ODF's? I have my idea in txt-files

**Comment by eturpault** [comment]
Hello @25Tim,
Yes this seems to be interesting, and in the same type of processing as what I have done with OdfEdit.
I am going to deliver soon OdfEdit with the compass extension feature, then I have no other planned implementation, why not to work on your idea. What is it more precisely ?

**Comment by eturpault** [comment]
Hello Tim,
If I understand well your txt files, the idea should be to copy samples data from a HW ODF to another HW ODF (I have not understood well according to which criteria).
But I am sorry, I am not willing to implement a program for such a special need, and which is doing modifications in a HW ODF (which may cause me problems from Milan Digital Audio for having done a SW which is generating or reworking a HW ODF).

**Comment by 25Tim** [comment]
That's right. Sampleset manufacturers also have their own crib and software for HW-ODF's and here is a site with HW-software https://organ.monespace.net/organworks/software.html . the problem is, I think, not so much Milan but the sample set manufacturers, it does not allow everyone, you have to pay* attention to their licenses, that's why you also have an HBX files. (*not correct from translator)
I can ask Milan Digital Audio for this project
Friesach is easy to understand this for search objects RankID -1 have StopID-1 have SampleID-10001 to SampleID-15330
then Rank2 - stopid2 - SampleID 20001 to SampleID 25330 in Pedal
in Pipe_SoundEngine01 is this for Rank1: 
 first               <PipeID>101</PipeID>         and last: <PipeID>132</PipeID
		<RankID>1</RankID>
I must look in more ODF's.

**Comment by 25Tim** [comment]
Of course, you should only use it for CODM created ODF's or composite sets, as far as it is allowed, not for original ODF's

**Comment by 25Tim** [comment]
Oh yes on the page is pipeTune that unfortunately doesn't work, otherwise I would have tried that. I can understand you, otherwise I will look for another programmer. Thank you very much anyway. but still clarify the legal.

**Comment by eturpault** [comment]
Indeed there are SW permitting to generate HW ODF from scratch, like at the link that you have given.
In your case the idea is to extract data from HW ODFs generated by sample set manufacturers to build another HW ODF. In the ["Terms and Conditions" of Piotr Graboswki](https://piotrgrabowski.pl/terms-and-conditions/) for example I am not sure it is allowed to rework or make an extract of a sample set.
So I am still not in favor to implement a SW permitting to make automatical manipulations and moves of data from official HW ODFs.

About the content of a HW ODF, you can use OdfEdit to navigate among the objects of an ODF:
- open an HW ODF with OdfEdit
- it will generate a GO ODF that you can ignore
- at the right side of OdfEdit click on the tab "HW sections"
- inside this tab, you can navigate in the HW ODF objects by tree view, double-clicking on a parent or a child object of the current selected object (in red) to make it the new current object and see its parents/children. When you click on an object, you can see its attributes in the text editor at the left side of the tree, presented in a GO ODF manner (attribute name=attribute value).

For example you can navigate from the top object _General to Division then Stop then StopRank then Rank then Pipe then Pipe layer then Attack/Release samples then Sample, and come back in the tree.

**Comment by 25Tim** [comment]
thank you ok, i have odfedit, I already know the program almorse.org (Site is deleted), les Deutsch and Andrea Bonzi have make composite ODF's. I can ask P. Grabowski to.  your OdfEdit make the same for GO-ODF's. 
I have not yet thought of this legal situation with the data copy from the ODF. My interest is that it is legal, thanks.
My Composite is for GO ready and works good and PitchTuning is good, this have i make for HW with no more pitch informations i have samples are not work correct and on other the pitch is not good. Hence my idea to copy the data by hand it takes forever
